feature/gribi/mpls_in_udp/otg_tests/mpls_in_udp_scale/mpls_in_udp_scale_test.go

```diff
@@ -0,0 +1,1757 @@
+	var nhOps []*client.OpResult
+
+	// Build next-hops for both ports
+	for i, p := range []string{"port3", "port4"} {
+		port := dut.Port(t, p)
+
+		switch {
+		case deviations.GRIBIMACOverrideWithStaticARP(dut):
+			nh, op := gribi.NHEntry(
+				nhIDs[i], "MACwithIp", deviations.DefaultNetworkInstance(dut),
+				fluent.InstalledInFIB,
+				&gribi.NHOptions{Dest: []string{otgPort3DummyIP.IPv4, otgPort4DummyIP.IPv4}[i], Mac: magicMac},
+			)
+			nhEntries = append(nhEntries, nh)
+			nhOps = append(nhOps, op)
+
+		case deviations.GRIBIMACOverrideStaticARPStaticRoute(dut):
+			nh, op := gribi.NHEntry(
+				nhIDs[i], "MACwithInterface", deviations.DefaultNetworkInstance(dut),
+				fluent.InstalledInFIB,
+				&gribi.NHOptions{Interface: port.Name(), Mac: magicMac, Dest: magicIP},
+			)
+			nhEntries = append(nhEntries, nh)
+			nhOps = append(nhOps, op)
+
+		default:
+			nh, op := gribi.NHEntry(
+				nhIDs[i], "MACwithInterface", deviations.DefaultNetworkInstance(dut),
+				fluent.InstalledInFIB,
+				&gribi.NHOptions{Interface: port.Name(), Mac: magicMac},
+			)
+			nhEntries = append(nhEntries, nh)
+			nhOps = append(nhOps, op)
+		}
+	}
+
+	// Build NHG with both next-hops (ECMP)
+	nhMap := map[uint64]uint64{
+		nhIDs[0]: 1, // weight 1
+		nhIDs[1]: 1, // weight 1
+	}
+	nhg, nhgOp := gribi.NHGEntry(nhgID, nhMap, deviations.DefaultNetworkInstance(dut), fluent.InstalledInFIB)
+	nhEntries = append(nhEntries, nhg)
+	nhOps = append(nhOps, nhgOp)
+
+	// Install all NH + NHG entries
+	c.AddEntries(t, nhEntries, nhOps)
+
+	// Add IPv6 route for outer destination to point to the NHG
+	c.AddIPv6(t, outerIPv6Dst+"/128", nhgID,
+		deviations.DefaultNetworkInstance(dut),
+		deviations.DefaultNetworkInstance(dut),
+		fluent.InstalledInFIB)
+
+	t.Logf("Installed ECMP route %s/128 via ports 3 and 4", outerIPv6Dst)
+}
+
+// programMPLSinUDPEntries programs gRIBI entries for MPLS-in-UDP encapsulation.
+// It installs a single NextHop that performs MPLS-in-UDP encapsulation with the
+// provided outer IPv6 and UDP header attributes, associates multiple NextHopGroups
+// (NHGs) with that NextHop, and finally installs IPv6 /128 routes pointing to each NHG.
+func programMPLSinUDPEntries(t *testing.T, dut *ondatra.DUTDevice, mplsNHID, mplsLabelStart uint64, numNHGs int, outerIPv6Src, outerIPv6Dst string, outerDstUDPPort uint16, outerIPTTL, outerDSCP uint8) []fluent.GRIBIEntry {
+	t.Helper()
+	entries := []fluent.GRIBIEntry{
+		// Create the single MPLS-in-UDP NextHop
+		fluent.NextHopEntry().
+			WithNetworkInstance(deviations.DefaultNetworkInstance(dut)).
+			WithIndex(mplsNHID).
+			AddEncapHeader(
+				fluent.MPLSEncapHeader().WithLabels(mplsLabelStart),
+				fluent.UDPV6EncapHeader().
+					WithSrcIP(outerIPv6Src).
+					WithDstIP(outerIPv6Dst).
+					WithDstUDPPort(uint64(outerDstUDPPort)).
+					WithIPTTL(uint64(outerIPTTL)).
+					WithDSCP(uint64(outerDSCP)),
+			),
+	}
+	// Create NextHopGroups pointing to the same NH
+	for i := 0; i < numNHGs; i++ {
```

Comment:
Since `i` is starting from 0, you can use the condensed format here.

https://go.dev/play/p/1vjrIWQpx0f

```suggestion
	for i := range numNHGs {
```